### PR TITLE
Player background color and Sliding Up Panel transparency

### PR DESCRIPTION
**Pull request by BrightDV**
Hello,

In this PR, I added some UI improvements :
- First, I replaced, in dark mode, the bottom color of the gradient in the player screen from black to the `darkMutedColor` of the artwork. I didn't add it to the light mode as there were color issues with the buttons…
Player Screen with gradient
- I also added a translucent background for the `SlidingUpPanel`, in order to get it less aggressive if the background color is bright.
Before with a no-transparent-black background color:
Now Playing SlidingUpPanel in dark mode
After:
Now Playing SlidingUpPanel in dark modeNow Playing SlidingUpPanel in lightmode 

The opacity in light mode is higher as the opacity in dark mode because some components were hard to read.

Should I add settings to enable or disable these changes?
I will reformat the code and revert changes that I made in the project files,  to make the changes that I made easier to compare.
Hope it will work, also I'm sorry for all the commits “Autosave”, I used an online IDE et pushed from there to GitHub…

### CODE REVIEW

**Comment by Sangwan5688** [comment]
Well, it looks cool. I haven't checked the code, will check later. First of all, make sure to provide a switch button in settings for that. Secondly, does it look good for all different kind of images? It shouldn't be the case that it's looking good for some whereas worse for others. Also, there might be cases where `darkMutedColor` might be `null` or same as `dominantColor` so take actions for them accordingly. As for the sliding panel, umm I think you can do something better instead of changing opacity. You can try using different kind of color gradient there or you can try adding blur there. Blur will look much better than reduced opacity. Maybe try using dominant color with blur. And finally, it looks like there's some formatting issue which is causing build to fail, so just fix it

**Comment by BrightDV** [comment]
### Checklist :
- [x]  First of all, make sure to provide a switch button in settings for that. 
Will push the commit that introduce the switch for the background color.
- [ ]  Secondly, does it look good for all different kind of images? It shouldn't be the case that it's looking good for some whereas worse for others. 
Tested for the moment with ~150 songs and seems ok (tried for JioSaavn, YouTube, and local) but waiting more time.
- [ ] Also, there might be cases where `darkMutedColor` might be null or same as `dominantColor `so take actions for them accordingly. 
Partially done [here](https://github.com/BrightDV/BlackHole/blob/d6ad7a4da9c2400b451f3d215b7dcca3c02f10f3/lib/Helpers/dominant_color.dart#L30) : if `dominantColor` [is null](https://github.com/BrightDV/BlackHole/blob/d6ad7a4da9c2400b451f3d215b7dcca3c02f10f3/lib/Helpers/dominant_color.dart#L28) or  `darkMutedColor` [is null](https://github.com/BrightDV/BlackHole/blob/d6ad7a4da9c2400b451f3d215b7dcca3c02f10f3/lib/Helpers/dominant_color.dart#L29), the color is black by default. So if there is only `darkMutedColor` that is not null, the screen will use it as the dominant color. However, I didn't implement any fallback if the two colors are the same, for the moment.
- [ ] As for the sliding panel, umm I think you can do something better instead of changing opacity. You can try using different kind of color gradient there or you can try adding blur there. Blur will look much better than reduced opacity. Maybe try using dominant color with blur.
Nice, will try to add it instead!
- [ ] And finally, it looks like there's some formatting issue which is causing build to fail, so just fix it.
Yes, the IDE that I used have some troubles in formatting...

**Comment by BrightDV** [comment]
So I have tried to add your suggestions!
For the blur of the `SlidingUpPanel`, I kept the transparency and it looks nice for me

Maybe a higher opacity will look better?
I have tried with the dominant, but there were some problems with the colors of the other artworks.

Also, for the moment, there is one issue I have to fix: when coming from miniplayer, the `darkMutedColor` (bottom of the `LinearGradient`) is black because it doesn't have saved the previous state.
Besides, I removed all the random stuff and made a better formatting, packages have been updated too.

Finally, I fixed a small bug: it was possible to scroll behind the Now Playing section because of the size of the `Container` of the Now Playing `Text`. It was a fixed value and I just replaced it by the `nowplayingBoxHeight` [here.](https://github.com/Sangwan5688/BlackHole/pull/253/files#diff-ae47859bddfce5f2e006c55d65e05d5023fc88960962f1ae2558d48231c5e2f8L2107)

**Comment by Sangwan5688** [comment]
Cool, I will check the code and let you know if there's some issue or not

**Comment by BrightDV** [comment]
> And I don't know why the check isn't running, maybe closing and reopening pr will fix the issue

Maybe it is in [this commit](https://github.com/Sangwan5688/BlackHole/commit/f19d9b615c6df80cc09ddc29c268222d816ac1ce), because the pull requests have been removed from the workflow file?

Also, for the moment, there is one issue: when coming from miniplayer, the `darkMutedColor` is black because it isn't saved whereas the `gradientColor` is saved. I have tried to change the `gradientColor` to a list but after that, the colors were shown a song after the good one. (When playing song A, colors were black, and when skiping to song B, the background colors were changed to the A's colors.). I don't know if it is really clear...

**Comment by BrightDV** [comment]
So it was easier to implement than I thought… I just change `gradientColor` to a List and save in it with dominant & darker colors.
I have tried to fix the build issues, hope it will work!

File: lib/Screens/Player/audioplayer.dart [hunk 1]
```diff
@@ -2104,7 +2137,7 @@ class NameNControls extends StatelessWidget {
                 }
               },
               child: Container(
-                height: 50,
+                height: nowplayingBoxHeight,
```
Comment (Sangwan5688) [inline lib/Screens/Player/audioplayer.dart:2140]: Let it remain 50. There was some issue with `nowplayingBoxHeight`. The thing is this `header` height can't be more than 50 or something (don't know exact number). So if you increase container height more than header height then `now playing` text will overlap with the song list.

**Review by Sangwan5688 — COMMENTED** [review]
Cool, everything looks fine. One more thing have you added autoscroll? I don't see it anywhere in this PR

File: lib/Screens/Player/audioplayer.dart (hunk 1 above)
Comment (BrightDV) [reply lib/Screens/Player/audioplayer.dart:2141]: Ok, I reverted it.

**Comment by BrightDV** [comment]
> One more thing have you added autoscroll? I don't see it anywhere in this PR

Not in this PR. This one is only about the UI of the player's background gradient.
I have some ideas for other PRs (more small fixes) and I will retry to implement the autoscroll.

**Comment by Sangwan5688** [comment]
Cool, thanks for PR :)

**Pull request merged**